style_guide.rst

```diff
@@ -0,0 +1,393 @@
+Style Guide
+===========
+
+
+Introduction
+------------
+
+This style guide is intended to provide some basic coding
+conventions for programming assignments and projects in C. These conventions
+are intended to improve the readability of your code. From
+your perspective as a student, the readability of your code
+is important for two big reasons:
+
+#. If you have a problem with your code, or have trouble
+   debugging a particular issue, it will be easier for us
+   to provide assistance (and to provide it quickly) if your
+   code is easy to read.
+#. Writing readable code makes the lives of our graders easier.
+   Remember: a happy grader is a generous grader. Additionally,
+   graders *will* take points off for code that is hard to read
+   (which includes, but is not limited to, code that doesn't
+   follow the conventions in this style guide)
+
+This document is not intended to be a comprehensive C style guide.
+When you need guidance on a style issue not specified here,
+we recommend using the `Linux Kernel Style Guide <https://www.kernel.org/doc/html/v4.10/process/coding-style.html>`_
+(however, when that guide contradicts our guide, our guide will
+be considered the normative document).
+
+Code layout
+------------
+
+Indentation
+^^^^^^^^^^^
+
+You must follow one of the following two indent styles:
+
+* `K&R style <https://en.wikipedia.org/wiki/Indentation_style#K&R>`_ (you can use any of the variants listed on that page, except BSD KNF), or
+* `Allman style <https://en.wikipedia.org/wiki/Indent_style#Allman_style>`_
+
+Make sure you choose one indent style and use it **consistently**. Please note that you can use `astyle <http://astyle.sourceforge.net/>`_ to take existing code and
```

Comment:
Given that students are working on larger teams here, and then combining code as a whole group, would it be worth just picking one indent style for the class?

Reply:
The idea is that this guide will eventually be shared by multiple classes, so I think this is something that we would communicate outside this guide (i.e., "the style guide provides two indentation styles, but in this class we will be using X")